src/WebForms/Compilation/BaseTemplateParser.cs

```diff
@@ -67,6 +67,11 @@ protected internal Type GetUserControlType(string virtualPath)
     internal Type GetUserControlType(VirtualPath virtualPath)
     {
         Type t = GetReferencedType(virtualPath, false /*allowNoCompile*/);
+        if (PagesConfig.UserControlTypesDict.ContainsKey(virtualPath.FileName))
```

Comment:
please remove - I don't want to access the type this way

Reply:
what's the alternative @twsouthwick ? Even I think this is not correct way.. we can't pass TypedReference or can't pass null...

Reply:
i'm not sure but this is not the way. Hence why I'm requesting incrementally adding things so we can isolate out what the sticky points are

Reply:
resolving as the type interface is implementd